### PR TITLE
Watch nodes but filter events + some documentation and testing changes

### PR DESCRIPTION
**Pull request by shanalily**
Fix watching of nodes (which have many events). I wasn't able to use a predicate struct with filters the way I wanted to so I'm relying instead on a field in the reconcile struct which stores when nodes where last updated (i.e. had reconcile function run for them, whether they had new tags/labels or not).

I added the configurable minimum interval between updates back in. This can be configured in the options ConfigMap with field "interval".

### CODE REVIEW

**Comment by juan-lee** [comment]
Please update the PR/commit title to conform to the guidance here https://chris.beams.io/posts/git-commit/

File: azure/provider_id_test.go [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+		expected   string
+	}{
+		{
+			"",
+			"",
+		},
+	}
+
+	for _, tt := range parseProviderIDTest {
+		// do stuff
```
Comment (juan-lee) [inline azure/provider_id_test.go]: did you mean to add this test in this PR?

File: controller/config.go [hunk 2]
```diff
@@ -15,6 +16,7 @@ const (
 	DefaultLabelPrefix         string = "azure.tags"
 	DefaultTagPrefix           string = "node.labels"
 	DefaultResourceGroupFilter string = "none"
+	DefaultInterval            string = "5m" // interval between updates
```
Comment (juan-lee) [inline controller/config.go]: Needing a comment is sometimes an indication that the variable could have a more descriptive name.

File: azure/provider_id_test.go (hunk 1 above)
Comment (shanalily) [reply azure/provider_id_test.go]: That was an accident, I didn't realize I'd merged with the branch this was on.

File: controller/config.go (hunk 2 above)
Comment (shanalily) [reply controller/config.go]: MinSyncPeriod maybe? Since SyncPeriod in the manager struct is the max period.

**Pull request merged**